arch/xtensa/src/common/xtensa_swint.c

```diff
@@ -103,7 +103,7 @@ int xtensa_swint(int irq, void *context, void *arg)
       case SYS_save_context:
         {
           DEBUGASSERT(regs[REG_A3] != 0);
-          memcpy((uint32_t *)regs[REG_A3], regs, (4 * XCPTCONTEXT_REGS));
+          memcpy(*(uint32_t **)regs[REG_A3], regs, XCPTCONTEXT_SIZE);
```

Comment:
don't need change

Reply:
Should revert the change.

Reply:
We will have the same issue with CP state as in the restore state, but I see your comment about fpu test above.